python/az/aro/azext_aro/custom.py

```diff
@@ -129,10 +130,18 @@ def aro_list_credentials(client, resource_group_name, resource_name):
 
 def aro_update(client, resource_group_name, resource_name, worker_count=None,
                no_wait=False):
+    current = client.get(resource_group_name, resource_name)
+
+    if len(current.worker_profiles) != 1:
+        raise CLIError("Can only perform update on a cluster with 1 worker profile. This cluster has %d" % len(current.worker_profiles))
+
     oc = v2019_12_31_preview.OpenShiftCluster(
-        # TODO: [0] should not be hard-coded
```

Comment:
Currently there can be only one worker profile, so I think it's fine to hard code the index. But I added a check above to make sure that the update command fails, if the older version of the client is being used with a cluster which has more profiles.

https://github.com/Azure/ARO-RP/blob/2a1d2208cc3047faef8e23bdcc14f016861c0aac/pkg/api/v20191231preview/openshiftcluster_validatestatic.go#L105-L107